src/Shared/CertificateGeneration/UnixCertificateManager.cs

```diff
@@ -1,11 +1,19 @@
 using System;
 using System.Collections.Generic;
 using System.Security.Cryptography.X509Certificates;
+using Microsoft.Extensions.Tools.Internal;
 
 namespace Microsoft.AspNetCore.Certificates.Generation
 {
     internal class UnixCertificateManager : CertificateManager
     {
+        private string CertificateStoreKey => $"aspnet-{Environment.UserName}";
+
+        private List<CertificateStore> _certificateStores;
+
+        private List<CertificateStore> CertificateStores
+            => _certificateStores ??= CertificateStoreFinder.FindCertificateStores();
```

Comment:
Can we be explicit here instead of dealing with a collection? This just makes the code harder to follow and there are no real benefits for abstracting this, since the code is not extensible and its not going to be prone to change.

Reply:
I think it's beneficial to have this.
It allows to write unit tests that work against a `CertificateStore`.
And having the collection makes explicit how something like `IsTrusted` is related to the different locations that can trust the certificate.

Reply:
> It allows to write unit tests that work against a `CertificateStore`.

We only use integration/manual tests for these types of things, since automation for these things is problematic on our CIs.

> And having the collection makes explicit how something like `IsTrusted` is related to the different locations that can trust the certificate.

I don't feel incredibly strong about this, however the reason that I don't like having a collection is that we are unlikely to add more stores than the ones we have now (at least frequently) and that it makes it harder to debug things or emit specific logs for a given store.